src/inc/clr/fs/path.h

```diff
@@ -55,35 +74,40 @@ namespace clr
             }
 
             //-----------------------------------------------------------------------------------------
-            static inline bool
-            HasUncPrefix(LPCWSTR wzPath)
-            {
-                _ASSERTE(!clr::str::IsNullOrEmpty(wzPath)); // Must check this first.
-                return wzPath[0] != W('\0') && wzPath[0] == W('\\')
-                    && wzPath[1] != W('\0') && wzPath[1] == W('\\')
-                    && wzPath[2] != W('\0') && wzPath[2] != W('?');
-            }
-
-            //-----------------------------------------------------------------------------------------
-            static inline bool
-            HasDrivePrefix(LPCWSTR wzPath)
-            {
-                _ASSERTE(!clr::str::IsNullOrEmpty(wzPath)); // Must check this first.
-                return wzPath[0] != W('\0')
-                    && wzPath[1] != W('\0') && wzPath[1] == W(':')
-                    && ((wzPath[0] >= W('a') && wzPath[0] <= W('z')) ||
-                        (wzPath[0] >= W('A') && wzPath[0] <= W('Z')));
-            }
-
-            //-----------------------------------------------------------------------------------------
             // Returns true if wzPath represents a relative path.
             static inline bool
             IsRelative(LPCWSTR wzPath)
             {
-                _ASSERTE(!clr::str::IsNullOrEmpty(wzPath)); // Must check this first.
-                return !HasLongFormatPrefix(wzPath)
-                    && !HasUncPrefix(wzPath)
-                    && (!HasDrivePrefix(wzPath) || wzPath[2] != W('\\'));
+                _ASSERTE(wzPath != nullptr);
```

Comment:
Not in the paths where this function is currently called, but for the purposes of this function, an empty string is considered a relative path. The old code checking for relative path allowed for an empty string, and so does System.IO.Path.IsRelative(), so wanted to make it consistent with those.